addons/analytic/models/analytic_plan.py

```diff
@@ -244,38 +245,71 @@ def _get_applicability(self, **kwargs):
 
     def unlink(self):
         # Remove the dynamic field created with the plan (see `_inverse_name`)
+        if not self:
+            return super().unlink()
         self._find_plan_column().unlink()
         return super().unlink()
 
-    def _find_plan_column(self):
-        return self.env['ir.model.fields'].sudo().search([
-            ('name', 'in', [plan._strict_column_name() for plan in self]),
-            ('model', '=', 'account.analytic.line'),
-        ])
+    def _get_plan_column_models(self):
+        # Return the models on which the plan column should be created
+        return ['account.analytic.line']
```

Comment:
If you used a mixin like suggested above, this could be
```suggestion
    def _get_plan_column_models(self):
        return self.env.registry.descendants(['analytic.column.mixin'], '_inherit') - {'analytic.column.mixin'}
```

No need to override/extend